packages/core/ios/Sources/JsRefreshRate.m

```diff
@@ -0,0 +1,104 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+// This ensures we can use [_bridge dispatchBlock]
+#import <React/RCTBridge+Private.h>
+#import "JsRefreshRate.h"
+
+typedef void (^frame_time_callback)(double frameTime);
+
+@implementation JsRefreshRate {
+  CADisplayLink *_jsDisplayLink;
+  NSTimeInterval _lastFrameTimestamp;
+  frame_time_callback _frameTimeCallback;
+  BOOL _isStarted;
+}
+
+@synthesize bridge = _bridge;
+// This ensures we can use [_bridge dispatchBlock]
+RCT_EXPORT_MODULE()
+
+// The plugin is initialized twice: the first time by React (as we registered this with RCT_EXPORT_MODULE),
+// and that is the moment the bridge is set.
+// The second time in DdSdk when we start it. This singleton pattern ensures we only have one instance running and
+// it is initialized.
+static JsRefreshRate *_pluginSingleton = nil;
+- (instancetype)init {
+  self->_isStarted = false;
+
+  if (!_pluginSingleton) {
+    self = [self initSingleton];
+    _pluginSingleton = self;
+  }
+
+  return _pluginSingleton;
+}
+
+- (instancetype)initSingleton {
```

Comment:
Could you explain what would be the benefits of doing so?

Reply:
This prevents creating two separate instances when calling the singleton from two separate threads.

The rest is just a naming convention that's more popular among ios devs which is `sharedInstance`.

Reply:
We have 2 conflicting comments here :) https://github.com/DataDog/dd-sdk-reactnative/pull/315#discussion_r1021306409

My proposal does not ensure thread safety of accessing `_shared`, but from RN you will always access it from the `mainDispatchQueue`, right? up to you to decide which one best based on how `JSRefreshRate` usage might evolve.
What's important tho is that you split the instance init and singleton accessor:
```objective-c
/// static accessor to singleton
+ (JsRefreshRate *)shared;

/// instance constructor
- (instancetype)init;
```
Let me know if you need further explanation on objc convention 👍

Reply:
For now it will always be accessed from the same thread, but it might change in React Native, and I don't wish to debug this in months if that's the case ^^
I'll include @maciejburda's suggestion, it makes sense to me to add this extra 👍
I'll also add a comment to explain, so we can remember why it's done this way.